Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -1178,13 +1180,32 @@ void RenderBlockFlow::determineLogicalLeftPositionForChild(RenderBox& child, App
         startPosition += (style().isLeftToRightDirection() ? 1 : -1) * verticalScrollbarWidth();
     if (style().scrollbarGutter().bothEdges && !isHorizontalWritingMode())
         startPosition += (style().isLeftToRightDirection() ? 1 : -1) * horizontalScrollbarHeight();
-    LayoutUnit totalAvailableLogicalWidth = borderAndPaddingLogicalWidth() + availableLogicalWidth();
+    const LayoutUnit totalAvailableLogicalWidth = borderAndPaddingLogicalWidth() + availableLogicalWidth();
+
+    auto computeSelfAlignedLogicalLeftPositionForChild = [&](LayoutUnit currentPosition) -> LayoutUnit {
```

Comment:
If we don't need this function until later, shouldn't we declare it later?

Reply:
That's true ... I defined it up here because the WebKit style seems to be to use default lambda capture `[&]` and I wanted to minimize the number of objects I was capturing (not that it would actually affect performance, but to reduce the cognitive load of knowing what's accessible in the lambda)

Reply:
I think the algorithm just is easier to read if the lambdas are defined right before they're used.

Reply:
Agreed, I made that change :)